### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Unlie9**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    db_author = crud.get_author(db, author_id=author.id)
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Author already exists"
+        )
+    return crud.create_author(db=db, author=author)
```
Comment (DevSpaciX) [inline main.py]: Provide return type annotations in all places

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

File: alembic/env.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+if config.config_file_name is not None:
+    fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+target_metadata = [Base.metadata]
```
Comment (Arsen-hrynevych) [inline alembic/env.py]: target_metadata = Base.metadata

File: database/crud.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    db_book = models.Book(
+        title=book.title,
+        summary=book.summary,
+        publication_date=book.publication_date,
+        author_id=book.author_id,
+    )
+    db.add(db_book)
+    db.commit()
+    db.refresh(db_book)
+    return db_book
```
Comment (Arsen-hrynevych) [inline database/crud.py]: Where try except block?

File: database/engine.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
```
Comment (Arsen-hrynevych) [inline database/engine.py:4]: After imports always 2 blank line

File: database/models.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from sqlalchemy import Column, ForeignKey, Integer, String, Date
```
Comment (Arsen-hrynevych) [inline database/models.py]: If importing more than two items, it is better to write them on separate lines. This improves readability and maintenance of the code. 

Example:

From taxi.models import ( 
	Driver,
	Manufacturer,
	Car
)
Read more about topic [here](https://peps.python.org/pep-0008/#imports)

File: main.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import List, Optional
+
+from database import schemas, crud
+from database.engine import SessionLocal
+
```
Comment (Arsen-hrynevych) [inline main.py:12]: After imports always 2 blank line

File: main.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        db: Session = Depends(get_db)
+) -> schemas.Book:
+    author = crud.get_author(db=db, author_id=book.author_id)
+
+    if author is None:
+        raise HTTPException(
+            status_code=400,
+            detail='Author does not exist'
+        )
+    return crud.create_book(db=db, book=book)
```
Comment (Arsen-hrynevych) [inline main.py]: Doesn't follow dry principle

**Review by Arsen-hrynevych — CHANGES_REQUESTED** [review]
Placing schemas, models, and crud in the database folder is often done to organize code related to database operations. However, this structure can be misleading because:

Schemas: Represent data validation and serialization, typically used in API requests and responses, not limited to database interactions.

Models: Define the structure of your database tables, but they are also part of your application's core logic and are often used across different layers.

CRUD: Contains functions for database operations, but it's part of the business logic layer, not strictly tied to the database.

**Review by Arsen-hrynevych — COMMENTED** [review]
Thanks

**Review by Arsen-hrynevych — APPROVED** [review]
Thanks

**Pull request closed without merging**